### PR TITLE
update addon template description for health probe and custom sign ca

### PR DESCRIPTION
**Pull request by zhujian7**

### CODE REVIEW

File: enhancements/sig-architecture/82-addon-template/README.md [hunk 1]
```diff
@@ -36,8 +36,8 @@ straightforward addon development way.
 
 - The addon agent image is still required.
 - The hosted mode addon is not supported.
-- The pre-delete-hook feature is not supported.
 - For complicated addon which has special RBAC requirement etc would still need to use addon-framework.
+- For the agent runtime, only support Deployment, not support Daemonset, Statefulset, etc.
```
Comment (qiujian16) [inline enhancements/sig-architecture/82-addon-template/README.md:40]: I think you support that, but cannot return status? we should clarify this. A user can set sts/deploy in the same template, but addon status only reflect status of deploy right now.

Comment (zhujian7) [reply enhancements/sig-architecture/82-addon-template/README.md:40]: but we only inject volumes that can access the hub cluster into the deployment.

Comment (zhujian7) [reply enhancements/sig-architecture/82-addon-template/README.md]: patl

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**